kyo-prelude/shared/src/main/scala/kyo/Emit.scala

```diff
@@ -96,29 +96,47 @@ object Emit:
           */
         def apply[A: Flat, S](v: A < (Emit[V] & S))(using tag: Tag[Emit[V]], frame: Frame): A < S =
             ArrowEffect.handle(tag, v)(
-                handle = [C] => (input, cont) => cont(Ack.Stop)
+                handle = [C] => (input, cont) => cont(())
             )
     end RunDiscardOps
 
     inline def runDiscard[V >: Nothing]: RunDiscardOps[V] = RunDiscardOps(())
 
-    final class RunAckOps[V](dummy: Unit) extends AnyVal:
-        /** Runs an Emit effect, allowing custom handling of each emitted value via an Ack.
+    final class RunForeachOps[V](dummy: Unit) extends AnyVal:
+        /** Runs an Emit effect, allowing custom handling of each emitted value.
           *
           * @param v
           *   The computation with Emit effect
           * @param f
-          *   A function to process each emitted value and return an Ack
+          *   A function to process each emitted value
           * @return
           *   The result of the computation
           */
-        def apply[A: Flat, S, S2](v: A < (Emit[V] & S))(f: V => Ack < S2)(using tag: Tag[Emit[V]], frame: Frame): A < (S & S2) =
+        def apply[A: Flat, S, S2](v: A < (Emit[V] & S))(f: V => Unit < S2)(using tag: Tag[Emit[V]], frame: Frame): A < (S & S2) =
             ArrowEffect.handle(tag, v)(
                 [C] => (input, cont) => f(input).map(cont)
             )
-    end RunAckOps
+    end RunForeachOps
 
-    inline def runAck[V >: Nothing]: RunAckOps[V] = RunAckOps(())
+    inline def runForeach[V >: Nothing]: RunForeachOps[V] = RunForeachOps(())
+
+    final class RunWhileOps[V](dummy: Unit) extends AnyVal:
+        /** Runs an Emit effect, allowing custom handling of each emitted value with a boolean result determining whether to continue.
+          *
+          * @param v
+          *   The computation with Emit effect
+          * @param f
+          *   A function to process each emitted value
+          * @return
+          *   The result of the computation
+          */
+        def apply[A: Flat, S, S2](v: A < (Emit[V] & S))(f: V => Boolean < S2)(using tag: Tag[Emit[V]], frame: Frame): A < (S & S2) =
```

Comment:
I've been placing the computation as the last parameter group to make it work well with `pipe`: `computation.pipe(Emit.runForeach(v => ...))`. I'm not very sure about this design decision but it's the common pattern across the code base.

Reply:
Isn't that how it is here?

Reply:
no, `v` would come after `f`

Reply:
Done -- but this is not great for type inference in other use cases (see changes to test)